### ISSUE TITLE
Unexpected inference input and output

### ISSUE THREAD

**Issue by Robbie-Palmer** [NVIDIA/dl-inference-server#23]
I am trying to add my own TensorFlow graphdef model to my model repository (I can successfully run the example ResNet model)
Here is my config.pbtxt file which I based my config.pbtxt file on the examples in the [inference user guide](https://docs.nvidia.com/deeplearning/sdk/inference-user-guide/index.html#schema) and the ResNet example
```
name: "my_model"
platform: "tensorflow_graphdef"
max_batch_size: 1
input [
  {
    name: "input"
    data_type: TYPE_FP32
    # This model is channels first
    format: FORMAT_NCHW
    dims: [ 3, 512, 512 ]
  }
]
output [
  {
    name: "output"
    data_type: TYPE_FP32
    dims: [ 2, 512, 512 ]
  }
]
instance_group [
  {
    kind: KIND_GPU,
    count: 4
  }
]
```
When I start the inference server with this configuration it displays the error:
`Invalid argument: unexpected inference input 'gpu_0/data' allowed inputs are: input_1`

When I change the input name to "input_1" it displays the error:
`Invalid argument: unexpected inference output 'output' allowed outputs are: batch_normalization_1/Reshape_4, batch_normalization_1/Reshape_4/shape, batch_normalization_1/Reshape_5, batch_normalization_1/Reshape_5/shape, ...`
with a list of 374 options

So I am confused about what the names of the input and output sections are for and what they should be
In the [inference user guide configuration schema](https://docs.nvidia.com/deeplearning/sdk/inference-user-guide/index.html#schema) the TensorFlow graph def example just names them `input` and `output`; the other examples name them `data` and `prob` and the ResNet example names them `gpu_0/data` and `gpu_0/softmax`
None of these options are available in the allowed inputs or outputs the server lists
I randomly selected `block5_conv2/Relu` from the list to see what would happen and it loaded successfully, but I assume this is to do with the final layer of my model which is a softmax layer and there is no reference to softmax in the allowed output list

The guide says that we can find more information in the [Model Config Schema](https://github.com/NVIDIA/dl-inference-server/blob/master/src/core/model_config.proto) in this repo, but the name fields in the `ModelInput` and `ModelOutput` are just specified as strings with a default value
So I cannot find any documentation on what these values should be, or reference to the allowed values and what they mean/what the differences are between the options

**Comment by deadeyegoodwin** [comment]
The input and output name(s) for your model will be specific to your model. They are determined when you create the model. As you see the inference server gives you feedback on possible values. Unfortunately, graphdef is not a good model representation in that it doesn't explicitly say what the output(s) of the model are... basically any node in the graph can be an output, which is why the inference server is saying there are 374 possible outputs. Did you create the model yourself? Get it from someone? That person would need to tell you the "true" output(s) of the model. For a classification-type model it is usually called something like softmax or logits or similar... but that is not a requirement.

**Comment by deadeyegoodwin** [comment]
The entire TensorRT Inference Server has been open-sourced at https://github.com/NVIDIA/tensorrt-inference-server. 

This repo is no longer accepting changes and will soon be closed so please move all future work to https://github.com/NVIDIA/tensorrt-inference-server.
